roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -429,124 +429,123 @@ def get_property_data(self, property_name):
         return ret_val
 
     @staticmethod
-    def write_segmentation(segext_obj, savepath, metadata_dict, **kwargs):
+    def write_segmentation(segext_obj, savepath, metadata_dict=None, **kwargs):
+        source_path = segext_obj.filepath
+        print(f'writing nwb for {segext_obj.extractor_name}\n')
         if isinstance(metadata_dict, str):
             with open(metadata_dict, 'r') as f:
-                metadata_dict = yaml.safe_load(f)
+                metadata = yaml.safe_load(f)
 
+        metadata_dict = metadata
         # NWBfile:
         nwbfile_args = dict(identifier=str(uuid.uuid4()), )
-        if 'NWBFile' in metadata_dict:
-            nwbfile_args.update(**metadata_dict['NWBFile'])
+        nwbfile_args.update(**metadata_dict['NWBFile'])
         nwbfile = NWBFile(**nwbfile_args)
 
         # Subject:
-        if 'Subject' in metadata_dict:
-            nwbfile.subject = Subject(**metadata_dict['Subject'])
+        nwbfile.subject = Subject(**metadata_dict['Subject'])
 
         # Device:
-        if isinstance(metadata_dict['ophys']['Device'], list):
-            for devices in metadata_dict['ophys']['Device']:
+        if isinstance(metadata_dict['Ophys']['Device'], list):
+            for devices in metadata_dict['Ophys']['Device']:
                 nwbfile.create_device(**devices)
         else:
-            nwbfile.create_device(**metadata_dict['ophys']['Device'])
+            nwbfile.create_device(**metadata_dict['Ophys']['Device'])
 
         # Processing Module:
-        ophys_mod = nwbfile.create_processing_module('ophys',
+        ophys_mod = nwbfile.create_processing_module('Ophys',
                                                      'contains optical physiology processed data')
 
         # ImageSegmentation:
-        image_segmentation = ImageSegmentation(metadata_dict['ophys']['ImageSegmentation']['name'])
+        image_segmentation = ImageSegmentation(name=metadata_dict['Ophys']['ImageSegmentation']['name'])
         ophys_mod.add_data_interface(image_segmentation)
 
-        # OpticalChannel:
-        channel_names = segext_obj.get_channel_names()
-        input_args = [dict(name=i) for i in channel_names]
-        for j, i in enumerate(metadata_dict['ophys']['ImagingPlane']['optical_channel']):
-            input_args[j].update(**i)
-        optical_channels = [OpticalChannel(input_args[j]) for j, i in enumerate(channel_names)]
+        #OPtical Channel:
+        channel_names = [segext_obj.get_channel_names()]
+        input_args=[[dict(name=i) for i in channel_names[k]] for k in range(segext_obj.no_planes)]
```

Comment:
Shouldn't there me a check if this is a MultiSegmentationExtractor?

Reply:
yes, its fixed here 2ab62d4fa4dfaf0f7165f3b40358883f666485a6
The multisegmentation is implemented here #36